magda-typescript-common/src/Registry.ts

```diff
@@ -48,10 +50,31 @@ export default class Registry {
             .catch(createServiceError);
     }
 
+    async forEachRecord(callback: (record: Record) => Promise<any>, aspect?: Array<string>, optionalAspect?: Array<string>, dereference?: boolean) {
```

Comment:
I'd have preferred to use a generator/iterator style approach, but it seems it can't be done since there's promises involved (see [for-await-of proposal here](https://github.com/tc39/proposal-async-iteration)).